includes/wikia/models/DesignSystemGlobalNavigationModel.class.php

```diff
@@ -0,0 +1,287 @@
+<?php
+
+class DesignSystemGlobalNavigationModel extends WikiaModel {
+	const DEFAULT_LANG = 'en';
+
+	private $wikiId;
+	private $lang;
+
+	public function __construct( $wikiId, $lang = self::DEFAULT_LANG ) {
+		parent::__construct();
+
+		$this->wikiId = $wikiId;
+		$this->lang = $lang;
+	}
+
+	public function getData() {
+		global $wgUser;
+
+		$data = [
+			'logo' => [
+				'header' => [
+					'type' => 'link-image',
+					'href' => $this->getHref( 'fandom-logo' ),
+					'image' => 'wds-company-logo-fandom',
+					'title' => [
+						'type' => 'text',
+						'value' => 'Fandom powered by Wikia'
+					]
+				]
+			],
+			'search' => [
+				'module' => [
+					'type' => 'search',
+					'results' => [
+						'url' => $this->getPageUrl( 'Search', NS_SPECIAL, 'fulltext=Search' ),
+						'param-name' => 'query'
+					],
+					'suggestions' => [
+						'url' => WikiFactory::getHostById( $this->wikiId ) . '/index.php?action=ajax&rs=getLinkSuggest&format=json',
+						'param-name' => 'query'
+					],
+					'placeholder-inactive' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-search-placeholder-inac`tive'
+					],
+					'placeholder-active' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-search-placeholder-active'
+					]
+				]
+			],
+			'create_wiki' => [
+				'header' => [
+					'type' => 'link-text',
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'wikia-create-wiki-link-start-wikia'
+					],
+					'href' => GlobalTitle::newFromText( 'CreateNewWiki', NS_SPECIAL, 80433 )
+						->getFullURL( 'uselang=' . $this->lang ),
+				]
+			]
+		];
+
+		if ( $this->lang === 'en' ) {
+			$data[ 'verticals' ] = $this->getVerticalsSection();
+			$data[ 'wikis' ] = $this->getWikisSection();
+		} else {
+			$data[ 'community-central' ] = $this->getCommunityCentralLink();
+		}
+
+		if ( $wgUser->isLoggedIn() ) {
+			$data[ 'user' ] = $this->getLoggedInUserData( $wgUser );
+			$data[ 'notifications' ] = $this->getNotifications( $wgUser );
+		} else {
+			$data[ 'anon' ] = $this->getAnonUserData();
+		}
+
+		return $data;
+	}
+
+	private function getHref( $hrefKey ) {
+		return DesignSystemSharedLinks::getHref( $hrefKey, $this->lang );
+	}
+
+	private function getAnonUserData() {
+		return [
+			'header' => [
+				'type' => 'line-image',
+				'image' => 'wds-icons-user',
+				'title' => [
+					'type' => 'translatable-text',
+					'key' => 'global-navigation-userinfo-anon-avatar-title',
+				],
+			],
+			'links' => [
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "links" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)